plugins/out_s3/s3.c

```diff
@@ -1832,10 +1874,11 @@ static void cb_s3_upload(struct flb_config *config, void *data)
         m_upload = mk_list_entry(head, struct multipart_upload, _head);
         complete = FLB_FALSE;
 
-        if (m_upload->complete_errors >= MAX_UPLOAD_ERRORS) {
+        if (m_upload->complete_errors >= ctx->ins->retry_limit) {
+            strftime(create_time, 20, "%Y-%m-%d %H:%M:%S", localtime_r(&(m_upload->init_time), &now_time));
             flb_plg_error(ctx->ins,
-                          "Upload for %s has reached max completion errors, "
-                          "plugin will give up", m_upload->s3_key);
+                          "chunk tag=%s, create_time=%s cannot be retried",
+                          (char *)m_upload->tag, create_time);
             mk_list_del(&m_upload->_head);
```

Comment:
shared print function plz

Reply:
@matthewfala You mean one function for all warnings or one for each type? I think it should be one for each type of warning- as in each use case. So may be one for retry_warn, and one for retry_failed?

Reply:
Shucks, I was thinking just one function for all the warnings.